pkg/config/daemonset_config.go

```diff
@@ -0,0 +1,128 @@
+*/
+
+package config
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/klog/v2"
+
+	"github.com/juicedata/juicefs-csi-driver/pkg/k8sclient"
+)
+
+const (
+	DaemonSetConfigMapName = "juicefs-daemonset-config"
+	DefaultConfigKey       = "default"
+)
+
+// DaemonSetConfig represents the configuration for a DaemonSet deployment
+type DaemonSetConfig struct {
+	Enabled      *bool                `yaml:"enabled,omitempty"`     // Explicitly enable/disable DaemonSet for this StorageClass
+	NodeAffinity *corev1.NodeAffinity `yaml:"nodeAffinity,omitempty"`
+}
+
+// GetDaemonSetConfig retrieves the DaemonSet configuration for a given StorageClass
+// It first checks for a StorageClass-specific configuration, then falls back to default
+func GetDaemonSetConfig(ctx context.Context, client *k8sclient.K8sClient, storageClassName string) (*corev1.NodeAffinity, error) {
+	log := klog.NewKlogr().WithName("daemonset-config")
+	
+	// Try to get the ConfigMap
+	configMap, err := client.GetConfigMap(ctx, DaemonSetConfigMapName, Namespace)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.V(1).Info("DaemonSet ConfigMap not found, using no node affinity", 
+				"configMap", DaemonSetConfigMapName, "namespace", Namespace)
+			return nil, nil // No ConfigMap means no node affinity restrictions
+		}
+		return nil, fmt.Errorf("failed to get DaemonSet ConfigMap: %v", err)
+	}
+
+	// Try to get StorageClass-specific configuration
+	if configData, exists := configMap.Data[storageClassName]; exists {
+		log.V(1).Info("Found StorageClass-specific DaemonSet configuration", 
+			"storageClass", storageClassName)
+		return parseDaemonSetConfig(configData)
+	}
+
+	// Fall back to default configuration
+	if configData, exists := configMap.Data[DefaultConfigKey]; exists {
+		log.V(1).Info("Using default DaemonSet configuration for StorageClass", 
+			"storageClass", storageClassName)
+		return parseDaemonSetConfig(configData)
+	}
+
+	log.V(1).Info("No DaemonSet configuration found for StorageClass", 
+		"storageClass", storageClassName)
+	return nil, nil
+}
+
+// parseDaemonSetConfig parses the configuration string into a NodeAffinity
+func parseDaemonSetConfig(configData string) (*corev1.NodeAffinity, error) {
+	config := &DaemonSetConfig{}
+	if err := yaml.Unmarshal([]byte(configData), config); err != nil {
+		return nil, fmt.Errorf("failed to parse DaemonSet configuration: %v", err)
+	}
+	return config.NodeAffinity, nil
+}
+
+// LoadDaemonSetNodeAffinity loads node affinity for a StorageClass from ConfigMap
+// This is called when creating or updating a DaemonSet for mount pods
+func LoadDaemonSetNodeAffinity(ctx context.Context, client *k8sclient.K8sClient, jfsSetting *JfsSetting) error {
+	log := klog.NewKlogr().WithName("daemonset-config")
+	
+	// Skip if not using DaemonSet deployment
+	if !StorageClassShareMount || !StorageClassDaemonSet {
+		return nil
```

Comment:
Shouldn't this be an error? If this is called from elsewhere and it doesn't error the assumption will be that the state is set